scripts/utils/trading_strategy_helpers.js

```diff
@@ -378,35 +380,68 @@ const buildTransferApproveDepositTransaction = async function(
   targetTokenAddress,
   investmentTargetToken,
   artifacts,
-  web3
+  web3,
+  storeDepositsAsFile = false
 ) {
   const fleetSize = slaves.length
   assert(fleetSize % 2 == 0, "Fleet size must be a even number")
+  const deposits = []
 
   let fundingTransaction = []
   const FleetSizeDiv2 = fleetSize / 2
   for (const i of Array(FleetSizeDiv2).keys()) {
+    const deposit = 
+    {
+      amount: investmentStableToken.div(new BN(FleetSizeDiv2)).toString(),
+      tokenAddress: stableTokenAddress,
+      userAddress: slaves[i]
+    }
+    deposits.push(deposit)
     fundingTransaction = fundingTransaction.concat(
       await calculateTransactionForTransferApproveDeposit(
         masterSafeAddress,
-        stableTokenAddress,
-        slaves[i],
-        investmentStableToken.div(new BN(FleetSizeDiv2)),
+        deposit.tokenAddress,
```

Comment:
can we use ...deposit, if the deposit object is arranged differently? I would be required to be arranged like this: tokenAddress, userAddress, amount